src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -616,19 +616,18 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 	}
 
 	private async _activateTaskProviders(type: string | undefined): Promise<void> {
-		if (type && this._activatedTaskProviders.has(type)) {
-			return;
-		}
 		// We need to first wait for extensions to be registered because we might read
 		// the `TaskDefinitionRegistry` in case `type` is `undefined`
 		await this._extensionService.whenInstalledExtensionsRegistered();
-		this._log('Activating task providers ' + (type ?? 'all'));
+		if (type && !this._activatedTaskProviders.has(type)) {
+			this._log('Activating task providers ' + (type ?? 'all'));
+		}
 		const result = await raceTimeout(
 			Promise.all(this._getActivationEvents(type).map(activationEvent => this._extensionService.activateByEvent(activationEvent))),
 			5000,
 			() => console.warn('Timed out activating extensions for task providers')
 		);
-		if (result && type) {
+		if (type && result && !this._activatedTaskProviders.has(type)) {
```

Comment:
this `this._activatedTaskProviders.has` check doesn't really matter but can't hurt